lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -544,11 +546,13 @@ def _on_update_status(self, event: UpdateStatusEvent):
         if self.unit.is_leader():
             self.opensearch_exclusions.cleanup()
 
-            if (health := self.health.apply(wait_for_green_first=True)) not in [
+            if (health := self.health.apply(wait_for_green_first=False)) not in [
                 HealthColors.GREEN,
                 HealthColors.IGNORE,
             ]:
-                event.defer()
```

Comment:
This defer brings no real benefit, as another `update-status` will forcefully happen.

Reply:
Well, there are a few answers here:
1) Update status [can be set to arbitrarily long values or even disabled](https://juju.is/docs/juju/list-of-model-configuration-keys#heading--update-status-hook-interval), hence, we should never be doing here anything other than keeping the status
2) On an one year span, update-status will outpace any other type of event, if we start deferring them we may have multiple update-status happening at once
3) The update-status sets a time threshhold, if our hooks take too long, then we accumulate update-status. If our update-status is taking too long (e.g. because of deferred previous update-status) then we will have an endless loop

What is the benefit this deferral brings?